internal/provider/aws.go

```diff
@@ -61,11 +67,14 @@ var awsZones = map[string]string{
 
 func (p *AWSInputProvider) Provide() internal.ProviderValues {
 	zonesCount := p.zonesCount()
-	zones := p.zones()
 	region := DefaultAWSRegion
 	if p.ProvisioningParameters.Parameters.Region != nil {
 		region = *p.ProvisioningParameters.Parameters.Region
 	}
+	zones := AWSZones(region, p.ZonesProvider.RandomZones(pkg.AWS, region, zonesCount))
```

Comment:
`zonesCount` is computed based on hard-wired `DefaultAWSMultiZoneCount` - should it be so?